.github/workflows/nosqlbench-benchmark.yml

```diff
@@ -0,0 +1,73 @@
+name: NoSQLBench Daily Benchmark
+
+# TODO: Set a daily schedule once this workflow has been tested.
+on:
+  workflow_dispatch: # Enable manual runs
+
+env:
+  NOSQLBENCH_JAR_URL: https://github.com/nosqlbench/nosqlbench/releases/download/5.17.9-release/nb5.jar
+  NOSQLBENCH_JAR: /tmp/nb5.jar
+  NOSQLBENCH_WORKLOAD: ./.github/config/keyvalue_workload.yaml
+  NOSQLBENCH_DRIVER_CONFIG: ./.github/config/driver.conf
+
+jobs:
+  nosqlbench_benchmark:
+    runs-on: ubuntu-latest
+    strategy:
+      # Do not cancel other jobs in the matrix if one fails. This ensures that a
+      # failure in the 'devel' environment does not prevent the 'prod' benchmark from running.
+      fail-fast: false
+      matrix:
+        environment: [prod, devel]
+        java: [11]
+    outputs:
+      status: ${{ job.status }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up JDK ${{ matrix.java }}
+        uses: actions/setup-java@v4
+        with:
+          distribution: temurin
+          java-version: ${{ matrix.java }}
+      - uses: 'google-github-actions/auth@v2'
+        with:
+          credentials_json: '${{ secrets.SPANNER_CASSANDRA_ADAPTER_CICD_SERVICE_ACCOUNT }}'
+      - name: Download NoSQLBench
+        run: curl -L $NOSQLBENCH_JAR_URL -o $NOSQLBENCH_JAR
+      - name: Build with Maven
+        run: mvn clean package -DskipTests
+      - name: Start Adapter
+        env:
+          NOSQLBENCH_DATABASE_URI: ${{ secrets.NOSQLBENCH_DATABASE_URI }}
+          SPANNER_ENDPOINT: ${{ secrets.SPANNER_DEVEL_ENDPOINT }}
+          GOOGLE_SPANNER_CASSANDRA_LOG_SERVER_ERRORS: true
+          MATRIX_ENVIRONMENT: ${{ matrix.environment }}
+        run: |
+          chmod +x ./.github/scripts/start-adapter.sh
+          ./.github/scripts/start-adapter.sh
+      - name: Run NoSQLBench
+        run: |
+          java -jar $NOSQLBENCH_JAR -v start workload=$NOSQLBENCH_WORKLOAD driver=cqld4 localdc=datacenter1 \
+          cycles=50000 threads=5 host=127.0.0.1 port=9042 driverconfig=$NOSQLBENCH_DRIVER_CONFIG --progress console:10s
```

Comment:
how long will this take? I think we should run the benchmark >2hr

Reply:
This runs for about 100 seconds while it's in development, I will update it once it's set up to run daily. But, do we need such a long run now, since we are just checking pass/fail? From a cost perspective, is it worth running a job this long? I'm not sure how these jobs are billed.